docker-images/certbot/certbot.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+TEST="--test-cert"
+
+if [ -z "${DOMAIN}" ]; then
+  echo Make sure the DOMAIN environment variable points to the domain.
+  exit 1
+fi
+
+if [ -z "${EMAIL}" ]; then
+  echo Make sure the EMAIL environment variable points to the email.
+  exit 1
+fi
+
+if [ -z "${SECRET}" ]; then
+  echo Make sure the SECRET environment variable points to the k8s secret.
+  exit 1
+fi
+
+if [ -z "${PROD}" ]; then
+  echo Making a test certificate because PROD environment variable is not set.
+else
+  echo Making a valid certificate because PROD environment variable is set.
+  TEST=""
+fi
+
+while true; do
+  echo Waiting 2 minutes to avoid hitting rate limits
+  sleep 2m
+  if ./kubectl get secret "${SECRET}" -o 'jsonpath={.data}' | jq -r '.["tls.crt"]' | base64 -d | openssl x509 -checkend 2592000 -noout -in -; then
+    echo Certificate is valid for at least 30 days
+    sleep 15d
+  else
+    certbot certonly "${TEST}" --non-interactive --agree-tos -m "${EMAIL}" --dns-google -d '*.'"${DOMAIN}" --dns-google-propagation-seconds 120 && \
+    (./kubectl create secret tls "${SECRET}" --cert /etc/letsencrypt/live/"${DOMAIN}"/fullchain.pem --key /etc/letsencrypt/live/"${DOMAIN}"/privkey.pem --dry-run=client --save-config -o yaml | ./kubectl apply -f -) && \
+    echo Created and saved certificate
```

Comment:
Should this also update the cert in all other namespaces?

Reply:
This is done by the operator below

Reply:
To clarify - the operator listens on changes to Challenge. As soon as any challenge changes (or gets uploaded), the certificate will be propagated to that challenge namespace.

There's a slight race condition, if, for example, you create a challenge and make it public immediately, then the Ingress might be created before the LetsEncrypt certificate is available. If that ever happens, the ManagedCertiticate will be used instead.

Reply:
What if no challenge changes?
I.e. I host challenges for a long time without touching them and the certbot updates the certificate. The operator would never notice this, no?

Reply:
yea that's a problem.

https://github.com/google/kctf/pull/200/commits/565b66bb651a3bfa325c1d3bdb9a3f021fc3525f#diff-c8b71c089af554bf4958c42612d1cb8421169936d5b4d18355b595876374d4ac